scripts/mbedtls_dev/psa_information.py

```diff
@@ -97,22 +98,31 @@ def automatic_dependencies(*expressions: str) -> List[str]:
     return sorted(psa_want_symbol(name) for name in used)
 
 # Define set of regular expressions and dependencies to optionally append
-# extra dependencies for test case.
-AES_128BIT_ONLY_DEP_REGEX = r'AES\s(192|256)'
-AES_128BIT_ONLY_DEP = ["!MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH"]
+# extra dependencies for test case based on key description.
 
-DEPENDENCY_FROM_KEY = {
-    AES_128BIT_ONLY_DEP_REGEX: AES_128BIT_ONLY_DEP
-}#type: Dict[str, List[str]]
-def generate_key_dependencies(description: str) -> List[str]:
-    """Return additional dependencies based on pairs of REGEX and dependencies.
+# Skip AES test cases which require 192- or 256-bit key
+# if MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH defined
+AES_128BIT_ONLY_DEP_REGEX = r'AES\s(192|256)'
```

Comment:
Compiling a regex improves performance and avoids accidentally combining it with a string.
```suggestion
AES_128BIT_ONLY_DEP_REGEX = re.compile(r'AES\s(192|256)')
```

Same for `ECB_NO_PADDING_DEP_REGEX`.